src/router_core/terminus.c

```diff
@@ -76,30 +76,49 @@ void qdr_terminus_free(qdr_terminus_t *term)
 }
 
 
+// DISPATCH-1461: snprintf() is evil - it returns >= size on overflow.  This
+// wrapper will never return >= size, even if truncated.  This makes it safe to
+// do pointer & length arithmetic without overflowing the destination buffer in
+// qdr_terminus_format()
+//
+static inline int safe_snprintf(char *str, size_t size, const char *format, ...)
```

Comment:
should this function be in some place like src/aprinf.h so it can be used by everybody ?

Reply:
eventually, yes - there are many places were snprintf is used, we should review them.
Until then, this fixes the immediate need.